### ISSUE TITLE
Add a command to generate plugin deriver

### ISSUE THREAD

**Issue by bhanu951** [Chi-teck/drupal-code-generator#148]
Please add a command to generate plugin derivative extending **DeriverBase** class.

Currently there is no command to generate plugin derivative.

Hence we need to code same boilerplate code again and again.
Please Create a generator class to generate plugin derivative.

**Comment by chx** [comment]
I presume If you want to extend DeriverBase then you want to generate a _deriver_ and not a _derivative_. Especially because while I added support for plugin classes to define themselves as derivatives, it doesn't need generator support as those are just plugin classes with a special id. So I took the liberty to retitle the issue.

I guess it's worth doing it if we allow injecting dependencies, looking at ContainerDerivativeDiscoveryDecorator there's a special ContainerDeriverInterface to support that. 

We could start it as a copy of the controller which has very similar semantics.

**Comment by bhanu951** [comment]
@chx  Thanks for looking into it. What I meant was to have a command to generate something similar to [this](https://git.drupalcode.org/sandbox/Bhanu951-3103712/-/tree/2.x-dev/derivative_examples/src/Plugin/Derivative?ref_type=heads)

**Comment by chx** [comment]
Yes, indeed that's a deriver. I note the inconsistent naming in core, if you look at deriver classes you will find 34 which are called deriver and 47 which is called a derivative. Meanwhile the base class is `DeriverBase` and the interface is `DeriverInterface` so they are all derivers. What a mess.
